src/jdbc/cubrid/jdbc/jci/UStatement.java

```diff
@@ -130,7 +131,12 @@
 	public int result_cache_lifetime;
 	private boolean result_cacheable = false;
 	private UStmtCache stmt_cache;
+	private UStatementCacheData stmt_cache_data = null;
 
+	private List<UResultTuple[]> tuple_list = null;
+	private List<Integer> fetched_number = null;
+	private List<Integer> first_cursor = null;
```

Comment:
I think UStatement does not need a list of UResultTuple[]. UStatement should have only UResultTuple[], which is a part of the entire result set that has been fetched. Instead of UStaement, UStatementCache should manage the list of UResultTuple[].